File: LeetCode/leetcodeQuestion/69.js
var mySqrt = function(x) {
    let left = 1;
    let right = x;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        if (mid * mid === x) {
            return mid;
        } else if (mid * mid < x) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return right;  // right will hold the largest integer whose square is <= x
};

File: LeetCode/leetcodeQuestion/69.cpp
#include <iostream>
using namespace std;

int mySqrt(int x) {
    if (x == 0) return 0;  
    int left = 1, right = x, result = 0;

    while (left <= right) {
        int mid = left + (right - left) / 2;
        
        if (mid <= x / mid) {
            result = mid;  
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    
    return result;
}

int main() {
    int x;
    cout << "Enter a non-negative integer: ";
    cin >> x;
    
    cout << "The square root of " << x << " is " << mySqrt(x) << endl;
    return 0;
}
